kratos/utilities/geometry_utilities.cpp

```diff
@@ -16,13 +16,112 @@
 // External includes
 
 // Project includes
-
-// Include base h
 #include "utilities/geometry_utilities.h"
 #include "utilities/geometrical_projection_utilities.h"
 
 namespace Kratos
 {
+
+std::string GetGeometryName(const GeometryData::KratosGeometryType TypeOfGeometry)
+{
+    KRATOS_TRY;
+
+    // Using switch over map as the compiler warns if some enum values are not handled in the switch
+    switch(TypeOfGeometry) {
```

Comment:
Would it be possible to reuse the `string_geometry_map` in `specifications_utilities.h`?
Iterate over the values of the map and return the key. Fewer sources of the same list are easier to maintain.

Reply:
a switch statement is much more efficient than a map

Reply:
I would refactor the specification utils after this PR

Reply:
Well yes. But I assume this function is not part of the program’s critical path due to its string-based logic. I’m curious about the speed vs maintainability tradeoff in this case. Or am I wrong in my assumptions?

I do not intend to re-open any PR, but merely asking questions and trying to understand.

Reply:
That is a good question, I cannot answer that without proper profiling

This is usually where @matekelemen steps in to amaze us with his micro-benchmarking skills 😄

Reply:
I'm with @huhaas on this one; correctness and maintainability is much more important here than performance. By the way, the more important question that pops into my mind is why getting the name of an object is not part of the object's (`Geometry`) or  the enum's (`KratosGeometryType`) features, but that of some obscure utility in a completely different place?

It doesn't make much sense, but I'd hate to disappoint @philbucher, so here goes:

The fact that we're using `std::string` instead of `std::string_view` for string literals way overshadows any performance benefit we may get from a `switch` statement instead of a linear search on a map:
- [switch <=> map comparison with std::string](https://quick-bench.com/q/xnVgNyvoUSJgCv3W53K6DD1seDc)
- [switch <=> map comparison with std::string_view](https://quick-bench.com/q/U2Warypfnsb62Zkjo7OWRI_otio)

Reply:
@matekelemen that website is very fun to use 😄 

Appended your example with two cases: [link](https://quick-bench.com/q/KESr40mqwSejkCi1QpG_XSsrO7o)
(the regular map can be either a bi-map or two maps (A->B and B->A) constructed from a single source)).
The problem size is too small for any specialized container. With N=20 the overhead is considerable.

Having the name be part of the object (OO style) will be a much bigger code rewrite. Then the enum construction will change and every usage of it, which are a lot.